internal/hscan/structmap.go

```diff
@@ -0,0 +1,79 @@
+package hscan
+
+import (
+	"reflect"
+	"strings"
+	"sync"
+)
+
+// structField represents a single field in a target struct.
+type structField struct {
+	index int
+	fn    decoderFunc
+}
+
+// structFields contains the list of all fields in a target struct.
+type structFields struct {
+	m map[string]*structField
+}
+
+// structMap contains the map of struct fields for target structs
+// indexed by the struct type.
+type structMap struct {
+	m sync.Map
+}
+
+func newStructMap() *structMap {
+	return new(structMap)
+}
+
+func (s *structMap) get(t reflect.Type) *structFields {
+	if v, ok := s.m.Load(t); ok {
+		return v.(*structFields)
+	}
+
+	fMap := getStructFields(t, "redis")
+	s.m.Store(t, fMap)
+	return fMap
+}
+
+func newStructFields() *structFields {
+	return &structFields{
+		m: make(map[string]*structField),
+	}
+}
+
+func (s *structFields) set(tag string, sf *structField) {
+	s.m[tag] = sf
+}
+
+func (s *structFields) get(tag string) (*structField, bool) {
+	f, ok := s.m[tag]
+	return f, ok
+}
+
+func getStructFields(t reflect.Type, fieldTag string) *structFields {
+	var (
+		num = t.NumField()
+		out = newStructFields()
+	)
+
+	for i := 0; i < num; i++ {
+		f := t.Field(i)
+
+		tag := f.Tag.Get(fieldTag)
+		if tag == "" || tag == "-" {
+			continue
+		}
```

Comment:
It looks we are ignoring fields that don't have `redis:"something"` tag. Is that intentional?

Usually we are supposed to generate default name from Go field name and only ignore the field when `redis:"-"`. What do you think?

https://pkg.go.dev/github.com/codemodus/kace could be used to generate default name.

Reply:
this parser is too simple....can only do the most basic work,  it needs to continue to improve its functions,
i can continue to improve it this week,,,if needed

Reply:
> It looks we are ignoring fields that don't have redis:"something" tag. Is that intentional?

Yeah. Explicitly defining what's to be scanned is better (fewer gotchas) in my opinion. Opting in rather than opting out with `"-"`.

> this parser is too simple....can only do the most basic work, it needs to continue to improve its functions,

It only handles primitive types. The right way to extend it with the `Decode()` interface which was in the original PR. I hope @vmihailenco has plans to introduce that functionality soon!